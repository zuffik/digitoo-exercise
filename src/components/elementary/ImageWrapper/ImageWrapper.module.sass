.root
  position: relative
  overflow: hidden
  & > img
    height: 100%
    margin: auto
    position: absolute
    left: -9999px
    top: -9999px
    right: -9999px
    bottom: -9999px
