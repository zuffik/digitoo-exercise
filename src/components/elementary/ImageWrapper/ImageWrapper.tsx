import * as React from 'react';
import styles from './ImageWrapper.module.sass'

interface Props {
    children: React.ReactNode;
    height: number;
}

export const ImageWrapper: React.FC<Props> = (props: Props) => {
    return (
        <div className={styles.root} data-testid="image-wrapper" style={{height: props.height + 'px'}}>
            {props.children}
        </div>
    );
};
