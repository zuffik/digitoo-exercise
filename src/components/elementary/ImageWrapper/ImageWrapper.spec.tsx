import * as React from 'react';
import {render} from '@testing-library/react'
import {ImageWrapper} from "./ImageWrapper";

describe('<ImageWrapper/>', () => {
    it('should render', () => {
        const {asFragment, getByTestId} = render(<ImageWrapper height={234}><img/></ImageWrapper>);
        expect(getByTestId('image-wrapper')).toHaveStyle('height:234px');
        expect(asFragment()).toMatchSnapshot();
    });
});
