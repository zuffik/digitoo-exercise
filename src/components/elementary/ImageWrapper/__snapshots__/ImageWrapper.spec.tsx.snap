// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ImageWrapper/> should render 1`] = `
<DocumentFragment>
  <div
    class="root"
    data-testid="image-wrapper"
    style="height: 234px;"
  >
    <img />
  </div>
</DocumentFragment>
`;
